src/ir/cost.h

```diff
@@ -789,13 +473,51 @@ struct CostAnalyzer : public Visitor<CostAnalyzer, Index> {
     return ret + visit(curr->left) + visit(curr->right);
   }
   Index visitSelect(Select* curr) {
-    return 2 + visit(curr->condition) + visit(curr->ifTrue) +
+    return 1 + visit(curr->condition) + visit(curr->ifTrue) +
            visit(curr->ifFalse);
   }
   Index visitDrop(Drop* curr) { return visit(curr->value); }
   Index visitReturn(Return* curr) { return maybeVisit(curr->value); }
   Index visitMemorySize(MemorySize* curr) { return 1; }
-  Index visitMemoryGrow(MemoryGrow* curr) { return 100; }
+  Index visitMemoryGrow(MemoryGrow* curr) { return 100 + visit(curr->delta); }
+  Index visitMemoryInit(MemoryInit* curr) {
+    return 6 + visit(curr->dest) + visit(curr->offset) + visit(curr->size);
+  }
+  Index visitMemoryCopy(MemoryCopy* curr) {
+    return 6 + visit(curr->dest) + visit(curr->source) + visit(curr->size);
```

Comment:
Some day we may want to have a "predictability" factor too, as copy (and call) may be much more, we can't tell. But probably not worth it in the near future.